src/Build/Construction/UsingTaskParameterGroupElement.cs

```diff
@@ -129,12 +129,15 @@ private static void VerifyCorrectParent(ProjectElementContainer parent)
             ProjectUsingTaskElement parentUsingTask = parent as ProjectUsingTaskElement;
             ErrorUtilities.VerifyThrowInvalidOperation(parentUsingTask != null, "OM_CannotAcceptParent");
 
-            // Now since there is not goign to be a TaskElement on the using task we need to validate and make sure there is a TaskFactory attribute on the parent element and 
+            // Now since there is not going to be a TaskElement on the using task we need to validate and make sure there is a TaskFactory attribute on the parent element and 
             // that it is not empty
             if (parentUsingTask.TaskFactory.Length == 0)
             {
                 ErrorUtilities.VerifyThrow(parentUsingTask.Link == null, "TaskFactory");
-                ProjectXmlUtilities.VerifyThrowProjectRequiredAttribute(parent.XmlElement, "TaskFactory");
+
+                // Parameters under UsingTask is now allowed even if TaskFactory is not specified on it.
+                // This is used to set Log or LogItemMetadata on individual parameters.
```

Comment:
Can probably just remove these three lines. I assume you made it not need TaskFactory so anyone could specify the two parameters you added?

Reply:
Right. Didn't want to be too destructive and wanted to leave a historical record here ;) This is one thing about this PR that might be potentially worrisome as we are effectively relaxing the XML schema a bit.